src/pytest_inmanta_lsm/remote_service_instance_async.py

```diff
@@ -24,6 +24,31 @@
 T = typing.TypeVar("T")
 
 
+def get_log_as_service_instance(log: model.ServiceInstanceLog, current: model.ServiceInstance) -> model.ServiceInstance:
+    """
+    This helper method allow to convert of a ServiceInstanceLog into the corresponding ServiceInstance.
+    It takes a ServiceInstance object as input to use it as a basis for the crafted instance.  Any field
+    which are shared with the ServiceInstanceLog object will be overwritten with the value from
+    ServiceInstanceLog object.
+
+    :param log: The ServiceInstanceLog to convert to a ServiceInstance object.
+    :param current: The ServiceInstance to use as a basis for the crafted instance.
+    """
+    try:
+        # Dump the log as a dict, to make it easier to get its attributes values
+        # programmatically
+        raw_log = log.model_dump()
+        return model.ServiceInstance(
+            **{attr: raw_log[attr] if attr in raw_log else value for attr, value in current.model_dump().items()}
+        )
```

Comment:
This looks OK to me, but if we are planning to add support for something like this natively to inmanta-lsm (for which @wouterdb is probably the better judge), then I don't think forward compatibility is a concern here (we could just call the method if it exists), so I think it would be both easier and cleaner to construct a `ServiceInstance` directly with the appropriate fields.

Reply:
I took the hint, it is true that it makes it a lot easier to read and understand, I also referenced the corresponding ticket I created in inmanta-lsm: https://github.com/inmanta/pytest-inmanta-lsm/pull/414/commits/350a631f513174d23d7df39014618215e94d5be2